lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -976,45 +977,10 @@ instance EncodeAddress n => ToJSON (ApiAddress n) where
     toJSON = genericToJSON defaultRecordTypeOptions
 
 instance ToJSON (ApiT DerivationIndex) where
-    toJSON (ApiT (DerivationIndex ix))
-        | ix >= firstHardened = toJSON (show (ix - firstHardened) <> "H")
-        | otherwise = toJSON (show ix)
-      where
-        firstHardened = getIndex @'Hardened minBound
-
+    toJSON = toJSON . toText . getApiT
```

Comment:
Serialization delegated to the `ToText` instance and moved near the data-type declaration.